flocker/node/agents/blockdevice.py

```diff
@@ -1472,7 +1550,9 @@ def from_state_and_config(discovered_dataset, desired_dataset):
         # Other node will need to detach first, but we need to wake up to
         # notice that it has detached until FLOC-3834 makes that info part
         # of cluster state. So we need to poll... but not too often:
-        Discovered.ATTACHED_ELSEWHERE: PollUntilAttached,
+        Discovered.ATTACHED_ELSEWHERE: Poll,
+        Discovered.REGISTERED: Poll,
```

Comment:
I suggest adding comments explaining why REGISTERED requires Poll, since it's not obvious (unless you know about AWS pecularities).